Remove CoffeeScript artifacts from codebase

boennemann opened the pull request:
- [x] multiline chaining calls again
- [x] remove use of `_ref`
- [x] use `i` rather than `_i` in for loops
- [x] don't use `__bind` magic
- [x] remove class magic (outer block): bpTap, bpView,
- [x] untangle ternaries
- [x] use angular helpers (`angular.isDefined() angular.isFunction`, etc.)
- [ ] more?!

boennemann commented:
@davidpfahler Here is an example of how it could be done e71a4a2

davidpfahler commented:
@boennemann So file based after all? I thought you wanted me to do it by language construct?

boennemann commented:
It's file based because I'm only handling the navbar on my branch.

davidnpma committed:
refactor: remove class magic in tap, view & config

davidnpma committed:
refactor(tap): remove CoffeeScript artifacts

davidnpma committed:
refactor(view): remove CoffeeScript artifacts

davidnpma committed:
test(view): getDirection should return null if segements do not match

Before:
/home/second -> /home/foo/bar should return 'normal' direction
/home/second/third -> /crazy/whatever should return 'reverse'

After:
/home/second -> /home/foo/bar should return null
/home/second -> /home/second/third should return 'normal'
/home/second/third -> /crazy/whatever should return null

davidnpma committed:
fix(view): getDirection should return null if segments do not match

BREAKING CHANGE:
getDirection method is now more strict about detecting the direction from URL segments.
It requires the URLs to be the same in but the one segment that they defer from one
another. Examples below illustrate URL changes and the returned value of getDirection
if fed with the respective URL segments.

Before:
/home -> /home/second returned 'normal'
/home/second -> /home returned 'reverse'
/home/second/third -> /home returned null
/home -> /home/second/third returned 'normal'
/home/second -> /crazy/bar/baz returned 'normal'

After:
/home -> /home/second returns 'normal'
/home/second -> /home returns 'reverse'
/home/second/third -> /home returns null
/home -> /home/second/third returns null
/home/second -> /crazy/bar/baz returns null

boennemann committed:
test(view): less strict direction detection

boennemann committed:
revert(view): less strict direction detection

boennemann committed:
test(view): use data from state to detect direction

boennemann committed:
feat(view): use data from state to detect direction

boennemann committed:
test(search): flush init timeout before testing resize

boennemann committed:
style: improve readability by reducing linelengths

boennemann committed:
test(tabbar): tab labels
